### PR TITLE
Adding issue and PR templates

### PR DESCRIPTION
**Pull request by ebeahan**
Implementing templates for PRs and issues in ECS. Blank issues are still allowed. Added config directing general ECS-related questions to the forums.

We will want to better align the templates/verbiage with the RFC process in the near future, however, I wanted to put fairly generic templates in place for now.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+name: Feature Request
+about: Request a new feature
+labels: "enhancement"
+
+--
+<!--
+Please first search existing issues for the feature you are requesting; it may already exist, even as a closed issue.
```
Comment (webmat) [inline .github/ISSUE_TEMPLATE/feature-request.md:8]: > even as a closed issue

I'd rather not mention the closed issues. I don't want to encourage comments on closed issues.

Mentioning to search first for their issue is good, though 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!--
+Thank you for your interest in and contributing to ECS! There are a
+few simple things to check before submitting your pull request that
+can help with the review process. You should delete these items from
+our submission, but they are here to help bring them to your attention.
+-->
+
+- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
+- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)?
+- If submitting code/script changes, have you verified all tests pass locally using `make check`?
```
Comment (webmat) [inline .github/PULL_REQUEST_TEMPLATE.md:10]:
```suggestion
- If submitting code/script changes, have you verified all tests pass locally using `make test`?
```

`make test` is much faster and doesn't modify files in place.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+- If submitting schema/fields updates, have you generated new artififacts and commited those changes?
```
Comment (webmat) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
- If submitting schema/fields updates, have you generated artifacts by running `make`, and committed those changes?
```

File: .github/ISSUE_TEMPLATE/bug.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+**Steps to reproduce**:
+
+Please include a *minimal* but *complete* recreation of the problem.
+
+1.
+2.
+3.
+
+**Provide logs (if relevant)**:
```
[the remark this answers is not in the document]
Comment (ebeahan) [reply .github/ISSUE_TEMPLATE/bug.md:25]: Makes good sense :-) 

I'm in favor of adding the additional issue template. I'll review some past issues to get an idea of what format/information is useful for schema-related issues.

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+name: Feature Request
+about: Request a new feature
+labels: "enhancement"
+
+--
+<!--
+Please first search existing issues for the feature you are requesting; it may already exist as an open issue.
+-->
+
+<!-- Describe the feature. -->
```
Comment (webmat) [inline .github/ISSUE_TEMPLATE/feature-request.md:11]: I think I would reword this one to cover both small additions (add a single missing field) as well as major additions (adding a whole field set).

So in terms of wording I wouldn't call this "feature request", but rather something like "schema changes and additions"?

And we could request details about what people want to propose changes to:

- context of the suggestion, why they're opening this issue
- field names
- example values for the fields
- suggested datatypes they think would be appropriate
- example events that should map to this use case

Comment (webmat) [reply .github/ISSUE_TEMPLATE/feature-request.md:11]: Eventually the RFC process will have special handling for new field set additions. But I think the above will be helpful to get the ball rolling in simple cases, as well as new field set additions.

In the context of the RFCs, we're simply asking for the first things we expect to touch on in [Stages 0 and 1](https://epixa.github.io/ecs/stages.html)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+can help with the review process. You should delete these items from
+our submission, but they are here to help bring them to your attention.
+-->
+
+- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
+- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)?
+- If submitting code/script changes, have you verified all tests pass locally using `make test`?
+- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes?
+- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
+- Have you added an entry to the [CHANGELOG](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)?
```
Comment (webmat) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: Perhaps make the name more obvious in the link text here. People sometimes still get confused between the two files
```suggestion
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)?
```

File: .github/ISSUE_TEMPLATE/schema-changes-additions.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+name: "Schema Changes and Additions"
+about: "Changes or additions to the ECS schema"
```
Comment (webmat) [inline .github/ISSUE_TEMPLATE/schema-changes-additions.md:3]:
```suggestion
about: "Changes or additions to ECS"
```

The S in ECS already stands for schema ;-)

File: .github/ISSUE_TEMPLATE/schema-issue.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+---
+name: Schema Issue
+about: "Report an issue with the ECS schema."
```
Comment (webmat) [inline .github/ISSUE_TEMPLATE/schema-issue.md:3]:
```suggestion
about: "Report an issue with ECS."
```

My brother, despite not being a software developer, understood recursion. He would say that DOS stood for the "DOS Operating System" 😂

File: .github/ISSUE_TEMPLATE/schema-changes-additions.md (hunk 7 above)
Comment (ebeahan) [reply .github/ISSUE_TEMPLATE/schema-changes-additions.md:3]: 🤦
